Setup preliminary navbar

rileysw opened the pull request:
* Created preliminary navigation bar with [Navbar](https://react-bootstrap.github.io/components/navbar/)
* Created custom `NavLink` and `PrivateNavLink` components to display public/private links based on users' login status
* Implemented custom active path checking with Next.js router
    * Added visuals/animations per Figma
* Replicated [spinning logo code](https://github.com/HackAtUCI/HackUCI-Site-2022/tree/develop/app/client/src/app/components/hack-logo) from archived HackUCI website

TODO: Integrate with login/logout endpoints when they are completed

rileysw committed:
setup initial navbar

taesungh reviewed (changes requested):
I appreciate the attempts to reincorporate designs from the previous years. We could simplify a lot of the effort though by working complementary to our libraries.

rileysw committed:
cleaned up and simplified code

rileysw committed:
changed navbar color to orange

taesungh committed:
Rebuild Navigation with `NavItem` list elements

- Add font weight 700 and increase spacing

taesungh committed:
Fix remaining issues with Navigation

- Reorganize style overrides
- Reword "Log In" and "Log Out" and specify `/logout` path
- Use next/link for navbar brand
- Fix `aria-expanded` status on navbar toggle
  - React Bootstrap does not handle this for us 😞

taesungh committed:
Fix props issue in `PrivateNavLinkItem`
